### DISCUSSION TITLE
Global CSS import on demand

### DISCUSSION THREAD

**Discussion by mycolaos** [vercel/next.js#12020] [Help]
The new build-in CSS and SASS support expects the global styles to be imported in the `pages/_app.js`. The reason is to avoid class name and ordering conflicts, https://nextjs.org/blog/next-9-3#built-in-sass-support-for-global-stylesheets.

Question, how the global styles should be imported on demand?

Example: using third part components that are loaded on demand and aren't used frequently. Makes no sense to download their stylesheets for the whole app.

**Comment by rafaelalmeidatk** [comment]
I tried to do it (migrating from next-css) but seems like you can't do it currently. You can read the reasons here: https://github.com/zeit/next.js/blob/canary/errors/css-npm.md#possible-ways-to-fix-it

**Comment by mycolaos** [comment]
I've tried to import the global CSS inside a `test.module.scss` SCSS stylesheet and it seems to work. Is it a legit way doing that? Shouldn't it be documented?

**Comment by ebs-PriyankaKanwar** [comment]
Do you want to request a feature or report a bug?
Bug

What is the current behavior?
CSS seems to be out of order in the "production" mode. And my style CSS has not loaded chunks in the production mode, and my CSS is out of order in a production build.

What is the expected behavior?
My style CSS has not loaded chunks in the production mode.
here is my build

First Load JS shared by all 107 kB
├ static/pages/_app.js 16.1 kB
├ chunks/1899c3d08f1d4a9ab6e4e22050e66cd1600464e8.b9ea19.js 3.53 kB
├ chunks/48183758.ede116.js 65 B
├ chunks/57a5805a715b88beb8aec3ac43d8f962ef8879cc.53c4e4.js 2.34 kB
├ chunks/9ed67fcaced240be487e7a2880bfff868999eacb.76e83a.js 8.29 kB
├ chunks/ab68c4ca07cacf4e3048ee2d6f69810106b4b489.68a8b3.js 10.4 kB
├ chunks/d32b6132cd0996134c9f87f7274f553f791cd563.18dde5.js 6.39 kB
├ chunks/d6e4d54ae3ad4e326943499c7720eb830c61721b.b7693f.js 4.41 kB
├ chunks/framework.4dd100.js 40.3 kB
├ runtime/main.d3f0e8.js 13.2 kB
├ runtime/webpack.361a6a.js 1.59 kB
├ css/16c4518f007189356065.css 54.9 kB
└ css/f93669f8200e9712e425.css 30.6 kB
Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.

Node: v13.7.0
Webpack:4.43.0
OS: Linux
Can you please help me.
Thanks

**Comment by dsumer** [comment]
You can provide it via a `<style>` tag which you include in a `<Head>` tag of the component which will be lazy loaded.
It's described in detail here: https://trueq.io/q/1#answer-2
